### PR TITLE
chore: auto update config docs

### LINKED ISSUE #243: [DOC] Auto Update Config Docs
### Change in the documentation

The configuration documentation is [currently manually maintained](https://github.com/open-feature/flagd/blob/main/docs/configuration/configuration.md). This can be automated in Cobra by using the [generator markdown command](https://github.com/spf13/cobra/blob/main/doc/md_docs.md). Automatic updates will ensure that the docs always reflect the latest configuration options.

### PR DESCRIPTION
**Pull request by skyerus**
## This PR

- Introduced `make generate-docs` which auto generates cmd config documentation
- Pipeline check that doc generation doesn't create a diff (fail if it does with a message to run `make generate-docs`)

### Related Issues

Fixes #243

### CODE REVIEW

File: docs/configuration/flagd.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+### Options
+
+```
+      --config string   config file (default is $HOME/.agent.yaml)
+  -x, --debug           verbose logging
+  -h, --help            help for flagd
+```
+
+### SEE ALSO
```
Comment (beeme1mr) [inline docs/configuration/flagd.md:13]: Did you mean to make this upper case?

File: docs/configuration/flagd_start.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+```
+
+### Options inherited from parent commands
+
+```
+      --config string   config file (default is $HOME/.agent.yaml)
+  -x, --debug           verbose logging
+```
+
+### SEE ALSO
```
Comment (beeme1mr) [inline docs/configuration/flagd_start.md:33]: Same question?

File: docs/configuration/flagd.md (hunk 1 above)
Comment (skyerus) [reply docs/configuration/flagd.md:13]: These files are generated by cobra

**Review by beeme1mr — APPROVED** [review]
Very cool!

**Review by toddbaert — APPROVED** [review]
Love it.

Checking it in the CI (therefor forcing it to be committed) is the best approach, I think.

**Pull request merged**